mmcls/models/necks/dolg.py

```diff
@@ -0,0 +1,219 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import Optional, Sequence
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from mmengine.model import BaseModule
+from mmengine.utils import is_seq_of
+
+from mmcls.registry import MODELS
+from .gem import GeneralizedMeanPooling
+
+
+class MultiAtrous(BaseModule):
+    """Multiple atrous convolution layers.
+
+    Args:
+        in_channel (int): Number of channels in the input.
+        out_channel (int): Number of channels in the output.
+        dilation_rates (Sequence[int]): The list of the dilation rates
+            of multiple atrous convolution. Defaults to `[3, 6, 9]`.
+        init_cfg (dict, optional): dictionary to initialize weights.
+            Defaults to None.
+    """
+
+    def __init__(self,
+                 in_channel: int,
+                 out_channel: int,
+                 dilation_rates: Sequence[int] = [3, 6, 9],
+                 init_cfg: Optional[dict] = None):
+        super(MultiAtrous, self).__init__(init_cfg)
+        assert is_seq_of(
+            dilation_rates,
+            int), ('``dilation_rates`` must be a sequence of int.')
+        assert out_channel % 2 == 0
+
+        # In the original implementation, here hidden-dim was set to a
+        # fixed 512 as it was designed for resnet101. We set it here to
+        # half of out_c for the sake of generality, which is consistent
+        #  when using resnet101.
+        hidden_channel = out_channel // 2
+        self.dilated_conv_list = nn.ModuleList([
+            nn.Conv2d(
+                in_channel,
+                hidden_channel,
+                kernel_size=(3, 3),
+                padding=rate,
+                dilation=rate) for rate in dilation_rates
+        ])
+
+        # global convolution
+        self.gap_conv = nn.Sequential(
+            nn.AdaptiveAvgPool2d(1),
+            nn.Conv2d(in_channel, hidden_channel, kernel_size=(1, 1)),
+            nn.ReLU())
+
+        # convolution after concat for smoothing
+        num = len(dilation_rates) + 1
+        self.conv_after = nn.Sequential(
+            nn.Conv2d(hidden_channel * num, out_channel, kernel_size=(1, 1)),
+            nn.ReLU())
+
+    def forward(self, x):
+        h, w = x.shape[2], x.shape[3]
+        local_feat_list = [
```

Comment:
There are some aspects that differ from some unofficial implementations, including this section. Is the accuracy reproduced?

https://github.com/dongkyuk/DOLG-pytorch/blob/main/model/dolg.py
https://github.com/ChristofHenkel/kaggle-landmark-2021-1st-place/blob/main/models/ch_mdl_dolg_efficientnet.py

Reply:
the same with https://github.com/dongkyuk/DOLG-pytorch/blob/9726243e8b150e429674f1f41ddacd79d215a2ec/model/dolg.py#L28 `nn.Upsample` just use `F.interpolate(mode='bilinear',)` 
 
I think it would be better to use the official implementation here.

The precision here needs to be supported by the dataset, so we won't reproduce the precision for now